tests/test.ml

```diff
@@ -27,11 +27,80 @@ let test_metrics () =
     "
     output
 
+
+(* "^[a-zA-Z_][a-zA-Z0-9_]*$" *)
+let valid_labels = [
+  "_";
+  "a";
+  "aA0b1B9c8C7z6Z5y4Y3x2X1";
+  "_______";
+]
+
+let invalid_labels = [
+  "";
+  "1";
+  "a bad label";
+]
+
+let check_valid_label label () =
+  let _l: LabelName.t = LabelName.v label in
+  ()
+
+let check_invalid_label label () =
+  try
+    let _l: LabelName.t = LabelName.v label in
+    failwith (label ^ " should be an invalid label")
+  with _ -> ()
+
+let test_valid_labels_set = List.map (fun label ->
+  label, `Quick, check_valid_label label
+) valid_labels
+
+let test_invalid_labels_set = List.map (fun label ->
+  label, `Quick, check_invalid_label label
+) invalid_labels
+
+let check_valid_metric metric () =
+  let _m: MetricName.t = MetricName.v metric in
+  ()
+
+let check_invalid_metric metric () =
+  try
+    let _m: MetricName.t = MetricName.v metric in
+    failwith (metric ^ " should be an invalid metric")
+  with _ -> ()
```

Comment:
Since we're currently doing `failwith (Fmt.strf "Invalid name %S" name)` and I routinely get told off by the compiler for pattern matching on a `Failure`, I don't think I can use `check_raises` this time. I could if we switched to a custom exception in future though.